src/Symfony/Component/Cache/Adapter/ProxyAdapter.php

```diff
@@ -40,18 +44,52 @@ public function __construct(CacheItemPoolInterface $pool, string $namespace = ''
             function ($key, $innerItem) use ($defaultLifetime, $poolHash) {
                 $item = new CacheItem();
                 $item->key = $key;
-                $item->value = $innerItem->get();
+                $item->value = $v = $innerItem->get();
                 $item->isHit = $innerItem->isHit();
                 $item->defaultLifetime = $defaultLifetime;
                 $item->innerItem = $innerItem;
                 $item->poolHash = $poolHash;
+                if (\is_array($v) && 1 === \count($v) && \is_array($v = $v[CacheItem::STATS_KEY] ?? null) && array(0, 1) === \array_keys($v)) {
+                    list($item->value, $item->stats) = $v;
+                } elseif ($innerItem instanceof CacheItem) {
+                    $item->stats = $innerItem->stats;
+                }
                 $innerItem->set(null);
 
                 return $item;
             },
             null,
             CacheItem::class
         );
+        $this->setInnerItem = \Closure::bind(
+            function (CacheItemInterface $innerItem, array $item) {
+                if ($stats = $item["\0*\0newStats"]) {
```

Comment:
Not sure it it'd help much, but maybe move `\0*\0` to `private const PREFIX = '\0*\0';` in this class?

Reply:
Not sure, using the const would just add indirection IMHO.
Other comments addressed thanks.